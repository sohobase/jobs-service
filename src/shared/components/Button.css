@import "../styles/colors.css";
@import "../styles/fonts.css";
@import "../styles/variables.css";

:root {
  --button-height: calc(3.6 * var(--unit));
  --button-offset: 0 var(--offset);
  --button-radius: calc(var(--button-height) / 2);
}

.button {
  position: relative;
  cursor: pointer;
  border: none;
  outline: 0;
  padding: var(--button-offset);
  height: var(--button-height);
  font-size: var(--font-size-normal);
  font-weight: var(--font-weight-semi-bold);
  line-height: var(--button-height);
  border-radius: var(--button-radius);

  &.accent {
    background-color: var(--color-accent);
    color: white;

    &:hover,
    &:active {
      background-color: color(var(--color-accent) blackness(10%));
    }
  }

  &:not(.accent) {
    background-color: var(--color-primary);
  }
}
