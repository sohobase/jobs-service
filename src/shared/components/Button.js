import classnames from 'classnames';
import React, { PropTypes } from 'react';
import style from './Button.css';

const Button = ({ accent, caption, className, large, onClick }) => {
  const classes = classnames(style.button,
    accent ? style.accent : undefined,
    large ? style.large : undefined,
    className,
  );

  return (
    <button className={classes} onClick={onClick}>
      {caption}
    </button>
  );
};

Button.propTypes = {
  accent: PropTypes.bool,
  caption: PropTypes.string,
  className: PropTypes.string,
  large: PropTypes.bool,
  onClick: PropTypes.func,
};

export default Button;
