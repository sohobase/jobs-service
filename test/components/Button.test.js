import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Button from '../../src/shared/components/Button';

describe('<Button />', () => {
  // -- Renders
  it('renders by default', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a caption property', () => {
    const tree = renderer.create(<Button caption="Test button" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('accepts a className property', () => {
    const tree = renderer.create(<Button className="my_button" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // -- Actions
  it('calls onClick property when clicked', () => {
    const onClick = jest.fn();
    const button = shallow(<Button onClick={onClick} />);
    button.simulate('click');
    expect(onClick).toBeCalled();
  });
});
