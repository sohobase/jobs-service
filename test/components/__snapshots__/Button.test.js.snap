exports[`<Button /> accepts a className property 1`] = `
<button
  className="button my_button"
  onClick={undefined} />
`;

exports[`<Button /> renders a caption property 1`] = `
<button
  className="button"
  onClick={undefined}>
  Test button
</button>
`;

exports[`<Button /> renders by default 1`] = `
<button
  className="button"
  onClick={undefined} />
`;
